drivers/storage/vbox/vbox_driver.go

```diff
@@ -0,0 +1,658 @@
+package vbox
+
+import (
+	"io/ioutil"
+	"net"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/akutz/gofig"
+	"github.com/akutz/goof"
+	vbox "github.com/appropriate/go-virtualboxclient/virtualboxclient"
+	"github.com/emccode/libstorage/api/registry"
+	"github.com/emccode/libstorage/api/types"
+	"github.com/emccode/libstorage/api/types/context"
+	"github.com/emccode/libstorage/api/types/drivers"
+	"github.com/emccode/libstorage/drivers/storage/vbox/executor"
+	"github.com/emccode/rexray/core/errors"
+)
+
+const (
+	// Name is the name of the driver.
+	Name = executor.Name
+)
+
+// Driver represents a vbox driver implementation of StorageDriver
+type Driver struct {
+	exec executor.Executor
+	sync.Mutex
+	config         gofig.Config
+	vbox           *vbox.VirtualBox
+	machine        *vbox.Machine
+	machineNameID  string
+	uname          string
+	passwd         string
+	endpoint       string
+	volumePath     string
+	useTLS         bool
+	controllerName string
+}
+
+func init() {
+	gofig.Register(executor.LoadConfig())
+	registry.RegisterStorageDriver(Name, newDriver)
+}
+
+func newDriver() drivers.StorageDriver {
+	return &Driver{}
+}
+
+// Name returns the name of the driver
+func (d *Driver) Name() string {
+	return Name
+}
+
+// Init initializes the driver.
+func (d *Driver) Init(config gofig.Config) error {
+	d.config = config
+	d.uname = d.config.GetString("virtualbox.username")
```

Comment:
I believe these should be accessed via the getters at the bottom. ie. d.username()